lib/stm32/common/spi_common_v3.c

```diff
@@ -0,0 +1,963 @@
+*/
+
+void spi_enable_tx_dma(uint32_t spi)
+{
+	SPI_CFG1(spi) |= SPI_CFG1_TXDMAEN;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief SPI Disable Transmit Transfers via DMA
+
+@param[in] spi Unsigned int32. SPI peripheral identifier @ref spi_reg_base.
+*/
+
+void spi_disable_tx_dma(uint32_t spi)
+{
+	SPI_CFG1(spi) &= ~SPI_CFG1_TXDMAEN;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief SPI Enable Receive Transfers via DMA
+
+This allows received data streams to proceed unattended using DMA to move data
+from the receive buffer as data becomes available. The DMA channels provided
+for each SPI peripheral are given in the Technical Manual DMA section.
+
+@param[in] spi Unsigned int32. SPI peripheral identifier @ref spi_reg_base.
+*/
+
+void spi_enable_rx_dma(uint32_t spi)
+{
+	SPI_CFG1(spi) |= SPI_CFG1_RXDMAEN;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief SPI Disable Receive Transfers via DMA
+
+@param[in] spi Unsigned int32. SPI peripheral identifier @ref spi_reg_base.
+*/
+
+void spi_disable_rx_dma(uint32_t spi)
+{
+	SPI_CFG1(spi) &= ~SPI_CFG1_RXDMAEN;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief SPI Standard Mode selection
+@details Set SPI standard Modes
+Mode | CPOL | CPHA
+---- | ---- | ----
+ 0   |  0   |  0
+ 1   |  0   |  1
+ 2   |  1   |  0
+ 3   |  1   |  1
+@param[in] spi Unsigned int32. SPI peripheral identifier @ref spi_reg_base.
+@param[in] mode Unsigned int8. Standard SPI mode (0, 1, 2, 3)
+@sa spi_set_clock_phase_0 spi_set_clock_phase_1
+@sa spi_set_clock_polarity_0 spi_set_clock_polarity_1
+*/
+
+void spi_set_standard_mode(uint32_t spi, uint8_t mode)
+{
+	uint32_t reg32;
+	if (mode > 3) {
+		return;
+	}
+
+    reg32 = SPI_CFG2(spi) & ~(SPI_CFG2_CPOL | SPI_CFG2_CPHA);
+	if(mode & 1) {
+        SPI_CFG2(spi) = reg32 | SPI_CFG2_CPHA;
+    }
+    if(mode & 2) {
+        SPI_CFG2(spi) = reg32 | SPI_CFG2_CPOL;
+    }
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Configure the SPI as Master.
+
+The SPI peripheral is configured as a master with communication parameters
+baudrate, frame format lsb/msb first, clock polarity and phase. The SPI
+enable, CRC enable, CRC next CRC length controls are not affected.
+These must be controlled separately.
+
+@param[in] spi Unsigned int32. SPI peripheral identifier @ref spi_reg_base.
+@param[in] clkdiv Unsigned int8. Baudrate clock divider 0 (div/2) - 7(div/128).
+@param[in] cpol bool. Clock polarity, false for 0, true for 1.
+@param[in] cpha bool. Clock Phase, false for 0, true for 1.
+@param[in] lsbfirst bool. Frame format lsb/msb first.
+@param[in] keepgpio bool. True to keep GPIO controlled by SPI block.
+@returns int. Error code.
+*/
+
+int spi_init_master(uint32_t spi, uint8_t clkdiv, bool cpol, bool cpha,
+		    bool lsbfirst, bool keepgpio)
```

Comment:
at least do we agree on this definition? should keepgpio be predefined here and modifiable by a pair of dedicated functions?